### ISSUE TITLE
PqHandle: another command is already in progress

### ISSUE THREAD

**Issue by PNixx** [amphp/postgres#41]
After 2-6 hours high load I try error:
```
another command is already in progress: /project/vendor/amphp/postgres/src/PqHandle.php:202
```
After this error, no request is answered. I need restart amphp-cluster for restore correct work.

```
Installed packages, channel pecl.php.net:
=========================================
Package Version State
ev      1.0.6   stable
pq      2.1.8   stable
raphf   2.0.1   stable
```

**Comment by enumag** [comment]
I ran into this issue as well. Fortunately it was in scenario where I could reproduce it reliably. So I added some dumps into PqHandle to see what was happening in the PqHandle instance that eventually failed.

I found out that the last operation prior to the failure was `PqHandle::release()`. So I checked `$this->handle->busy` boolean (IMPORTANT: don't confuse it with the `$this->busy` promise, that's a different variable) and sure enough it was still `true` when the release method was called.

So I tried to use `Loop::defer()` and when that didn't help even `Loop::delay(1, ...)` bus even in that callback it was still true.

From what I checked the SQL query that caused it seems to be '"DISCARD ALL"'.

In the end the only thing I could do to avoid the crash was to consider the state invalid and add this into `PqHandle::release()`:

```
if ($this->handle->busy) {
    $this->close();
}
```

**Comment by kelunik** [comment]
@enumag Do you have a reproduction script you can share?

**Comment by enumag** [comment]
I wish but it's not that easy. It happens in the middle of my data import.

**Comment by enumag** [comment]
I noticed [this line](https://github.com/amphp/postgres/blob/02d367fb2a8c8d870a0af9c5a767ce6752542658/src/PqHandle.php#L236) in the PqHandle code and try to apply it to my case. So I copied this line into the `PqHandle::release()` method and it seems to have helped. According to log the case of `$this->handle->busy` being true in the release method still happened however the application no longer crashed.
